tests/unit/models/package-info-cache-test.js

```diff
@@ -17,107 +19,121 @@ describe('models/package-info-cache.js', function() {
-      expect(devDependencyPackages).to.exist;
-      expect(Object.keys(devDependencyPackages).length).to.equal(8);
+        expect(nodeModules).to.exist;
+        expect(nodeModules.entries).to.exist;
+        expect(Object.keys(nodeModules.entries).length).to.equal(6);
+      });
     });
 
-    it('shows projectPackageInfo.devDependencyPackages + missing dependencies = project.devDependencies', function() {
-      let devDependencyPackages = projectPackageInfo.devDependencyPackages;
-      expect(devDependencyPackages).to.exist;
-      let devDependencyPackageNames = Object.keys(devDependencyPackages);
-
-      let devDependencies = projectPackageInfo.pkg.devDependencies;
-      expect(devDependencies).to.exist;
-      let devDependencyNames = Object.keys(devDependencies).sort();
+    describe('invalid project', function() {
+      let project, fixturifyProject;
+      before(function() {
+        // create a new ember-app
+        fixturifyProject = new FixturifyProject('simple-ember-app', '0.0.0', project => {
+          project.addAddon('some-ember-addon', 'latest');
+          project.addAddon('some-other-ember-addon', 'latest');
+        });
 
-      let errorArray = projectPackageInfo.errors.getErrors();
-      let error = errorArray[0];
-      expect(error.type).to.equal('dependenciesMissing');
+        fixturifyProject.writeSync();
 
-      let missingDependencies = error.data;
+        // remove the actual node_module for the one addon, this will make the project invalid
+        fs.removeSync(`${fixturifyProject.root}/simple-ember-app/node_modules/some-ember-addon`);
 
-      let packageAndErrorNames = devDependencyPackageNames.concat(missingDependencies).sort();
+        project = fixturifyProject.buildProjectModel(Project);
+        project.discoverAddons();
 
-      expect(packageAndErrorNames).to.deep.equal(devDependencyNames);
-    });
+        pic = project.packageInfoCache;
+        projectPackageInfo = pic.getEntry(`${fixturifyProject.root}/simple-ember-app`);
+      });
 
-    it('shows projectPackageInfo has 1 in-repo addon named "ember-super-button"', function() {
-      let inRepoAddons = projectPackageInfo.inRepoAddons;
-      expect(inRepoAddons).to.exist;
-      expect(inRepoAddons.length).to.equal(1);
-      expect(inRepoAddons[0].realPath.indexOf(`simple${path.sep}lib${path.sep}ember-super-button`)).to.be.above(0);
-      expect(inRepoAddons[0].pkg.name).to.equal('ember-super-button');
-    });
+      after(function() {
+        fixturifyProject.dispose();
+      });
 
-    it('shows projectPackageInfo has 7 internal addon packages', function() {
-      let internalAddons = projectPackageInfo.internalAddons;
-      expect(internalAddons).to.exist;
-      expect(internalAddons.length).to.equal(7);
-    });
+      it('validates projectPackageInfo', function() {
+        expect(projectPackageInfo).to.exist;
+        expect(projectPackageInfo.pkg).to.exist;
+        expect(projectPackageInfo.valid).to.be.true; // WAT?
```

Comment:
fix this

Reply:
The 'valid' flag should be true unless the package contains a direct error (like a missing index.js file or invalid 'paths' entry or like that).